salt/modules/yumpkg.py

```diff
@@ -130,7 +158,12 @@ def _repoquery(repoquery_args,
             '{0}'.format(comment)
         )
     else:
-        out = call['stdout']
+        if _yum() == 'dnf':
+            # Remove the epoch when it is zero to maintain backward compatibility
+            remove_zero_epoch = call['stdout'].replace('-0:', '-')
```

Comment:
Can you give an example of the dnf output that would need to be modified like this? I just want to make sure the use of ``str.replace()`` here doesn't replace something we don't want to replace. It is possible that an ``re.sub()`` may be preferable, but I'd need to know more before I could say for sure.